drivers/digital-io/max22196/max22196.c

```diff
@@ -422,7 +422,7 @@ int max22196_set_chan_cnt(struct max22196_desc *desc, uint32_t ch,
 	int ret;
 	uint8_t msb_byte, lsb_byte;
 
-	if (ch > MAX22196_CHANNELS - 1)
+	if (desc->chip_id == ID_MAX22194 || ch > desc->max_chn_nb - 1)
```

Comment:
What should this evaluate to? I would rewrite the expression as it's a bit hard to follow.

Reply:
Will try them splitting into two if conditions, since they are not related one with the other.